test/raven.test.js

```diff
@@ -2528,6 +2530,30 @@ describe('Raven (public API)', function() {
                 Raven.captureException(new Error('err'));
             });
         });
+
+        it('should use optionsFromException', function() {
+            var error = new Error('crap');
+            error.extra = {foo: 'bar'}
+
+            Raven._globalOptions.optionsFromException = function(ex, options) {
+                options = _.assign({}, options)
+
+                options.extra = _.assign({}, ex.extra, options.extra)
+
+                return options
+            }
+
+            this.sinon.stub(Raven, 'isSetup').returns(true);
+            this.sinon.stub(Raven, '_handleStackInfo');
+
+            Raven.captureException(error);
+
+            assert.isTrue(Raven._handleStackInfo.calledWith(
+                // sinon sandbox does not exposes match property before sinon 2.0
+                sinon.match.any,
```

Comment:
Could be replaced by `this.sinon.match.any` if Sinon is updated to `>= 2.0` (support added in https://github.com/sinonjs/sinon/pull/1076)